api/src/main/java/org/apache/iceberg/io/ParallelIterable.java

```diff
@@ -72,8 +73,8 @@ private ParallelIterator(Iterable<? extends Iterable<T>> iterables,
             }
           }).iterator();
       this.workerPool = workerPool;
-      // submit 2 tasks per worker at a time
-      this.taskFutures = new Future[2 * ThreadPools.WORKER_THREAD_POOL_SIZE];
+      // In default, we submit 2 tasks per worker at a time.
```

Comment:
Style: please remove personal pronouns. Using "we" or "I" makes documentation longer and less direct. Also, no need for punctuation in line comments.

Actually, I don't think that this comment needs to change at all. It looks perfectly fine without the changes.